lib/generators/boring/devise_jwt/install/install_generator.rb

```diff
@@ -0,0 +1,200 @@
+
+module Boring
+  module DeviseJwt
+    class InstallGenerator < Rails::Generators::Base
+      desc "add devise-jwt to the application"
+
+      class_option :model_name, type: :string, aliases: "-m",
+                   default: "User",
+                   desc: "Tell us the user model name which will be used for authentication. Defaults to User"
+      class_option :use_rails_credentials, type: :boolean, aliases: "-rc",
+                   desc: "Use Rails credentials for devise_jwt_secret_key"
+      class_option :revocation_strategy, type: :string, aliases: "-rs",
+                   enum: %w[JTIMatcher Denylist Allowlist],
+                   default: "Denylist",
+                   desc: "Tell us the revocation strategy to be used. Defaults to Denylist"
+
+      def verify_presence_of_devise_gem
+        gem_file_content_array = File.readlines("Gemfile")
+        devise_is_installed = gem_file_content_array.any? { |line| line.include?('devise') }
+
+        return if devise_is_installed
+
+        say "We couldn't find devise gem. Please configure devise gem and run the generator again!", :red
+
+        abort
+      end
+
+      def verify_presence_of_devise_initializer
+        return if File.exist?("config/initializers/devise.rb")
+
+        say "We couldn't find devise initializer. Please configure devise gem and run the generator again!", :red
+
+        abort
+      end
+
+      def verify_presence_of_devise_model
+        return if File.exist?("app/models/#{options[:model_name].underscore}.rb")
+
+        say "We couldn't find the #{options[:model_name]} model. Please provide correct model name", :red
+
+        abort
+      end
+
+      def add_devise_jwt_gem
+        say "Adding devise-jwt gem", :green
+        Bundler.with_unbundled_env do
+          run "bundle add devise-jwt"
+        end
+      end
+
+      def add_devise_jwt_config_to_devise_initializer
+        say "Adding devise-jwt config to devise initializer", :green
+        jwt_config = <<~RUBY
+          config.jwt do |jwt|
+            jwt.secret = #{devise_jwt_secret_key}
+            jwt.dispatch_requests = [
+              ['POST', %r{^/sign_in$}]
+            ]
+            jwt.revocation_requests = [
+              ['DELETE', %r{^/sign_out$}]
+            ]
+            jwt.expiration_time = 15.day.to_i
+          end
+        RUBY
+
+        inject_into_file "config/initializers/devise.rb",
+                          optimize_indentation(jwt_config, 2),
+                          before: /^end/
+      end
+
+      def configure_revocation_strategies
+        say "Configuring revocation strategies for #{options[:revocation_strategy]}", :green
+
+        case options[:revocation_strategy]
+        when "JTIMatcher"
+          config_jti_matcher_strategy
+        when "Denylist"
+          config_denylist_strategy
+        when "Allowlist"
+          config_allowlist_strategy
+        end
+      end
+
+      private
+
+      def devise_jwt_secret_key
```

Comment:
Let's make sure to let user know where we have configured these keys in the after installation message so they can change this if required and configure the key correctly.

You can take a look at rswag PR for the example.